modules/library/coverage/src/main/java/org/geotools/coverage/processing/operation/Extrema.java

```diff
@@ -148,24 +156,65 @@ protected boolean computeOnGeophysicsValues(ParameterValueGroup parameters) {
 			final RenderedOp result = (RenderedOp) data;
 			final Map<String, Object> synthProp = new HashMap<String, Object>();			
 
-			// get the properties
-			final double[] maximums = (double[]) result
-					.getProperty(GT_SYNTHETIC_PROPERTY_MAXIMUM);
-			final double[] minimums = (double[]) result
-					.getProperty(GT_SYNTHETIC_PROPERTY_MINIMUM);
-			Object property=result.getProperty(GT_SYNTHETIC_PROPERTY_MIN_LOCATIONS);
-			if((property instanceof List[]))
-				synthProp.put(GT_SYNTHETIC_PROPERTY_MIN_LOCATIONS, (List<int[]>[])property);
-			property=result.getProperty(GT_SYNTHETIC_PROPERTY_MAX_LOCATIONS);
-			if((property instanceof List[]))
-				synthProp.put(GT_SYNTHETIC_PROPERTY_MAX_LOCATIONS, (List<int[]>[])property);
-
-			// return the map
-			synthProp.put(GT_SYNTHETIC_PROPERTY_MINIMUM, minimums);
-			synthProp.put(GT_SYNTHETIC_PROPERTY_MAXIMUM, maximums);
+                        if (JAIExt.isJAIExtOperation("Stats")) {
```

Comment:
Why are the two code paths (jai and jai-xt) looking so different? Add a comment explaining please